### ISSUE TITLE
TURN server doesn't response TCP relay candidate

### ISSUE THREAD

**Issue by yokoyamy** [coturn/coturn#399]
I setup turnserver 3.2.5.9 for WebRTC

It responds candidates for `host` `srfix` `relay`.

I'd like have a `relay` candidate as `tcp`.

However,I only can have `udp` .

How can I have a `relay` candidate as `tcp`.

Here is turnserver.conf

    listening-port=80
    listening-ip=my-server-global-ip
    external-ip=my-server-global-ip
    lt-cred-mech
    userdb=/opt/turnserver/etc/turnuserdb.conf
    realm=my.server.domain
    stale-nonce
    no-udp
    no-stun

Here is Response from Trickle ICE tool

    Time	Component	Type	Foundation	Protocol	Address	Port	Priority
    0.015	1	host	2508812977	udp	10.2.1.17	53175	126 | 32542 | 255
    0.059	1	srflx	3607399481	udp	Client-Global-ip	53175	100 | 32542 | 255
    0.086	1	relay	1628315121	udp	my-server-global-ip	54043	1 | 32542 | 255
    0.116	1	host	3674902081	tcp	10.2.1.17	9	90 | 32542 | 255
    39.827	Done
    39.833

**Comment by AmarOk1412** [comment]
@yokoyamy you need to send a request with `REQUESTED-TRANSPORT` = tcp

**Comment by CoreyCole** [comment]
You can do this in the url I believe:
```
'url': 'turn:52.xx.xx.xx:3478?transport=tcp'
```

**Comment by Neustradamus** [comment]
@misi: Can you look?

**Comment by Neustradamus** [comment]
@misi: Can you look?

**Comment by cedricve** [comment]
same question.. not working.

**Comment by misi** [comment]
@cedricve  you need to send a request with REQUESTED-TRANSPORT = tcp WebRTC browsers don't requesting-transport  AFAIK. HTH

**Comment by misi** [comment]
Transport tcp does mean only between TURN client and TURN server, but it will allocate UDP port on server.

**Comment by cedricve** [comment]
thanks @misi, we are actually using a public TURN server (on a VPS), but the turn client is within a very strictly firewalled environment. We opened the TCP relay ports (2xxxx-6xxxxx) on the internal firewall, but that didnt work.

**Comment by misi** [comment]
@cedricve  Pubilc TURN is fine.  
Turn client(Dst:TCP turnserver:443) =={tcp}==Strict Firewall=={tcp}==Turn server----{udp}---Turn Peer
I guess you use 443 or port 80 with tcp ..

What do you mean internal firewall?

**Comment by cedricve** [comment]
Yes we use 443, but the TURN client is within a company network. So therefore I believe the (UDP) port range to the TURN server has to be allowed.

**Comment by misi** [comment]
@cedricve  You mean the peer is behind the same strict firewall..

**Comment by cedricve** [comment]
Exactly!

**Comment by cedricve** [comment]
@misi just found the Peer in the strict firewall, is not generating srflx or relay candidates. Only host candidate are generated. Is this expected bahaviour? (not having opened the UDP port range, and thus not able to allocate).
